src/nft-card.ts

```diff
@@ -73,25 +58,57 @@ export class NftCard extends LitElement {
     `
     }
 
+    /* User configurable properties */
+    @property({type: Boolean}) public horizontal = false
+    @property({type: String}) public contractAddress = ''
+    @property({type: String}) public tokenId = ''
+    @property({type: String}) public width = this.horizontal ? '670px' : '388px'
+    @property({type: String}) public height = this.horizontal ? '250px' : '560px'
+    @property({type: String}) public network: Network = Network.Main
+
+    @property({type: Object}) private asset!: OpenSeaAsset
+    @property({type: Object}) private traitData = {}
+    @property({type: String}) private account: string = ''
+    @property({type: String}) private flippedCard = false
+    @property({type: Object}) private provider: any // = ('web3' in window) ? window.web3.currentProvider : new Web3.providers.HttpProvider('https://mainnet.infura.io')
+    // @property({type: Object}) private web3: any = new Web3(this.provider)
+    @property({type: Object}) private seaport!: OpenSeaPort
+
+    // Card state variables
+    @property({type: Boolean}) private loading = true
+    @property({type: Boolean}) private isOwnedByAccount = false
+    @property({type: Boolean}) private isUnlocked: boolean = false
+    @property({type: Boolean}) private hasWeb3: boolean = false
+    @property({type: Boolean}) private isMatchingNetwork: boolean = false
+
+    private static getProvider() {
+        if (window.ethereum) {
+            return window.ethereum
+        } else if (window.web3) {
+            return window.web3.currentProvider
+        } else {
+            return new Web3.providers.HttpProvider('https://mainnet.infura.io')
```

Comment:
you have some things that check `if (this.provider)` and assume that means the user has web3 somewhere. but this getter will always return a provider. so this line should be removed i think

Reply:
This is a good point. Maybe I'll have it return `null` indicating that we don't have web3?